www/js/control/general-settings.js

```diff
@@ -290,9 +293,18 @@ angular.module('emission.main.control',['emission.services',
         $scope.getUsername();
     };
     $scope.getUsername = function() {
-      CommHelper.getUsername().then(function(results){
-        $scope.settings.username = results['username']
-      })
+      var username = localStorage.getItem('username');
+      if (username != null) {
+        $scope.settings.username = username;
+      } else {
+        CommHelper.getUsername().then(function(results){
+          ionic.show
```

Comment:
what does this line do?

Reply:
If for some reason the username is not yet stored in local storage (like for when people are upgrading versions to the new version that stores it into localStorage) then it's going to retrieve the username from the server to display in settings.

Reply:
yeah but what does `ionic.show` with no arguments do?

Reply:
OH no idea how that got there, definitely shouldn't be there.